recipe/recipe.yaml

```diff
@@ -0,0 +1,69 @@
+schema_version: 1
+
+context:
+  name: mamba-ssm
+  version: 2.2.5
+
+package:
+  name: ${{ name|lower }}
+  version: ${{ version }}
+
+source:
+  url: https://github.com/state-spaces/mamba/archive/refs/tags/v${{ version }}.tar.gz
+  sha256: 1e072841619af9c171709b2c6ff17daa0d9bf7caf0fcb083f5c11754b63c6cdb
+
+build:
+  number: 1
+  script:
+    env:
+      MAMBA_FORCE_BUILD: "TRUE"
+    content:
+      - sed -i.bak 's@"ninja"@#"ninja"@g' setup.py
+      - sed -i.bak 's@"ninja"@#"ninja"@g' pyproject.toml
+      - ${{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  skip:
+    # mamba-ssm requires CUDA (or ROCm) GPU
+    - cuda_compiler_version == "None" or not linux
+
+requirements:
+  build:
+    - ${{ compiler('c') }}
+    - ${{ compiler('cxx') }}
+    - ${{ compiler('cuda') }}
+    - ${{ stdlib('c') }}
+  host:
+    - cuda-version ==${{ cuda_compiler_version }}
+    - ninja
+    - packaging
+    - pip
+    - python
+    - pytorch * cuda*
+    - setuptools >=61.0.0
+    # Remove triton<3.2.0 pin when pytorch depends on triton>=3.4.0
+    # containing the https://github.com/triton-lang/triton/pull/6570 bugfix
+    - triton <3.2.0
+    - wheel
+  run:
+    - einops
+    - packaging
+    - python
+    - pytorch
+    - setuptools >=61.0.0
+    - transformers
+
+tests:
+  - python:
+      imports:
+        - mamba_ssm
+      pip_check: true
+
+about:
+  homepage: https://github.com/state-spaces/mamba
+  summary: Mamba state-space model
+  license: Apache-2.0
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - timkpaine
+    - weiji14
```

Comment:
Added myself to the recipe maintainer list if you don't mind! I'd be interested in adding aarch builds later for Grace Hopper systems (xref https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/7764).